v3_usm.go

```diff
@@ -249,6 +249,41 @@ func (sp *UsmSecurityParameters) Copy() SnmpV3SecurityParameters {
 	}
 }
 
+// Generate the new salting by incrementing the randomly seeded counter. Used for sever responses.
+func (sp *UsmSecurityParameters) GenSalt() {
+	// http://tools.ietf.org/html/rfc2574#section-8.1.1.1
+	// localDESSalt needs to be incremented on every packet.
```

Comment:
Isn't this already handled by `initPacket()`?

Reply:
I tried to use that instead, but it is a private method and not accusable by the external package `GoSNMPServer`. I tried looking for public methods that eventually called it, but it got too cumbersome. Probably exactly why @slayercat made the public method initially. 

If it wasn't for `localAESSalt` and `localDESSalt` being private, I could externally increment the salt. 

Just realized that `GenKey` need to use the mutex lock, one moment.

Reply:
`GenKey` now correctly locks the mutex before updating the `SecretKey`

Also here is how  both functions are used in GoSNMPServer currently: https://github.com/slayercat/GoSNMPServer/blob/e5e512ec64b25cec8820baa669d9a270b1705ab0/agentcontrol.go#L174-L181

Reply:
What I'm wondering is, why these methods need to be called explicitly. Rather than have whatever needs the salt/key automatically do these things.

My goal is to make sure there isn't a more ergonomic way to handle this.

Reply:
@SuperQ if `localAESSalt` and `localDESSalt` are made public, then the `GenSalt` and `GenKeys` methods would not be needed . The only issue is that the mutex lock cannot be leveraged, but an external lock could be used instead.

That might be a better stopgap as:
* `PrivacyKey` is already public so crypto artifacts are already accessible externally
*  Will give more time to come up with a more ergonomic way for `gosnmp` to offer server-side capabilities while still allowing an initial implementation to experiment with 
 
Just tell me if you want to go down this route and I'll make the changes.

Reply:
Yea, I'm not sure what the best solution is. I would love to see more server capabilities added directly to `gosnmp`.

I think keeping the data private and adding public functions to do the work is better due to the need to make sure we lock the data.